src/python/pants/backend/python/goals/export.py

```diff
@@ -141,7 +141,7 @@ async def export_virtualenv(
                     "venv",
                     "--pip",
                     "--collisions-ok",
-                    "--remove=all",
+                    "--remove=pex",
```

Comment:
This seems much more safe (much smaller scope for collisions), but could it still cause an issue if the same PEX is run twice at the same time?

Reply:
Hypothetically, yeah, since the `pex_pex` is written to a fixed path (see line 128). In practice no because we only run this path once per `export` (see export_virtualenv_for_targets() below).

We run the tool export below multiple times, but that disambiguates the pex_pex path. 

In 2.15.x those code paths are unified, and everything is disambiguated by resolve name. That said, it makes sense in 2.15.x to make it a truly unique name, so there's one less issue to worry about.

Reply:
No, since the PEX is in a sandbox.

Reply:
Hrm, not sure on that actually, but its a bit neither here nor there in the cherry-pick context. This is faithful to main and does not have the foot-gun nature of the prior issue. I'll finish up analysis and file / fix if need be on main separately.

Reply:
Oh wait, right, this removes the requirements.pex

Reply:
We remove the pex pex below, in the "rm -f" post-processing command.

Reply:
Yeah - so concurrent `./pants export` runs can suffer from the requirements.pex being nuked out from under them still as @stuhood pointed out. So that's maybe a bug, but its a bug inherent to the way the core export goal is designed currently afaict. I'll leave this for others to sort and proceed.

Reply:
But anyway, the requirements.pex is not in a sandbox either, that's why we `--remove` it. It's materialized under dist, so the post-processing command can run on it

Reply:
I'm going to knock up a change in main that makes this all easier to worry about. No need to cherry-pick that one (and it won't pick cleanly anyway).

Reply:
That would be excellent.

Reply:
https://github.com/pantsbuild/pants/pull/17317